worker/upgradesteps/manifold.go

```diff
@@ -68,40 +67,26 @@ func Manifold(config ManifoldConfig) dependency.Manifold {
 				return nil, err
 			}
 
-			// Get the machine agent's jobs.
-			// TODO(fwereade): use appropriate facade!
-			agentFacade, err := apiagent.NewState(apiConn)
-			if err != nil {
-				return nil, err
-			}
-			entity, err := agentFacade.Entity(tag)
-			if err != nil {
-				return nil, err
-			}
-			jobs := entity.Jobs()
-
-			// Get machine instance for setting status on.
-			// TODO(fwereade): use appropriate facade!
-			machinerFacade := apimachiner.NewState(apiConn)
-			machine, err := machinerFacade.Machine(tag)
-			if err != nil {
-				return nil, err
-			}
-
 			// Get upgradesteps completed lock.
 			var upgradeStepsLock gate.Lock
 			if err := context.Get(config.UpgradeStepsGateName, &upgradeStepsLock); err != nil {
 				return nil, err
 			}
 
+			// Get a reporter capable of setting machine status
+			// to indicate progress to the user.
+			reporter, err := config.Reporter(apiConn)
+			if err != nil {
+				return nil, err
+			}
 			return NewWorker(
 				upgradeStepsLock,
 				agent,
 				apiConn,
-				jobs,
+				agent.CurrentConfig().Jobs(),
```

Comment:
This doesn't appear to be valid, sadly. AFAICT, we don't update the jobs in agent.conf when they change, i.e. when promoting a machine agent to a controller. We query the API for jobs in the agent code - maybe jobs should be removed from agent.conf?

Reply:
re-instated jobs retrieval from entity.